src/foam/lib/csv/CSVSupport.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.lib.csv;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import foam.core.ClassInfo;
+import foam.core.PropertyInfo;
+import foam.core.FObject;
+import foam.lib.parse.*;
+import foam.dao.Sink;
+
+
+public class CSVSupport 
+  extends foam.core.ContextAwareSupport
+{
+  protected Parser headParser = new Parser() {
+    private Parser delegate = new Repeat(new CSVStringParser(), new Literal(","));
+    public PStream parse(PStream ps, ParserContext x) {
+      return ps.apply(delegate, x);
+    }
+  };
+
+  public void inputCSV(InputStream is, Sink sink, ClassInfo classInfo) {
+
+    try {
+      BufferedReader reader = new BufferedReader(new InputStreamReader(is, "UTF-8"));
+      String line = reader.readLine();
+      StringPStream ps = new StringPStream();
+      ps.setString(line);
+      ParserContext x = new ParserContextImpl();
+      x.set("X", getX());
```

Comment:
What is this for?

Reply:
Actually, PaserContext is useless in this case. I will delete it and pass null as ParserContext.

Reply:
I would keep it in case we add support for parsing FObjectProperties from a CSV file.